Core/GDCore/IDE/Events/EventsVariableReplacer.cpp

```diff
@@ -260,29 +300,97 @@ class GD_CORE_API ExpressionVariableReplacer
 
   bool RenameOrRemoveVariableOfTargetVariableContainer(
       gd::String& variableName) {
+    const auto *currentVariablesRenamingChangesetNode =
+        GetCurrentVariablesRenamingChangesetNode();
+    if (!currentVariablesRenamingChangesetNode) {
+      return false;
+    }
+    const auto &oldToNewVariableNames =
+        currentVariablesRenamingChangesetNode->oldToNewVariableNames;
     if (oldToNewVariableNames.count(variableName) >= 1) {
       variableName = oldToNewVariableNames.find(variableName)->second;
       hasDoneRenaming = true;
       return true;
-    } else if (removedVariableNames.count(variableName) >= 1) {
+    } else if (
+      // Only the root variable is checked for removing.
+      currentVariablesRenamingChangesetNode ==
+                   &variablesRenamingChangesetRoot &&
+               removedVariableNames.count(variableName) >= 1) {
       removedVariableUsed = true;
       return true;
     }
 
     return false;  // Nothing was changed or done.
   }
 
+  void RenameVariableAndVisitChild(gd::String &variableName,
+                                   ExpressionNode *childNode) {
+    // `variableName` is modified by
+    // `RenameOrRemoveVariableOfTargetVariableContainer`.
+    const gd::String oldVariableName = variableName;
+    RenameOrRemoveVariableOfTargetVariableContainer(variableName);
+    if (childNode) {
+      bool hasBeenPushed =
+          PushVariablesRenamingChangesetNodeForVariable(oldVariableName);
+      childNode->Visit(*this);
+      PopVariablesRenamingChangesetNode(hasBeenPushed);
```

Comment:
Why not making the `if` outside of the method, instead of passing hasBeenPushed or true every time?

Reply:
I guess this enforce the caller to provide the information and to be sure the check is made?

Reply:
Fine with me then!